lib/executor.js

```diff
@@ -316,10 +316,49 @@ function setupMiddleware(app, instructions) {
     }
     assert(typeof factory === 'function',
       'Middleware factory must be a function');
+    data.config = getUpdatedConfigObject(data.config, app);
     app.middlewareFromConfig(factory, data.config);
   });
 }
 
+function getUpdatedConfigObject(config, app) {
+
+  var DYNAMIC_CONFIG_PARAM = /\$\{(\w+)\}$/;
+  var updatedConfigObject = {};
+
+  function getConfigVariable(param) {
+    var configVariable = param;
+    var match = configVariable.match(DYNAMIC_CONFIG_PARAM);
+    if (match) {
+      configVariable = app.get(match[1]);
+      assert(configVariable, param + ' does not resolve to a valid object');
+    }
+    return configVariable;
+  }
+
+  function interpolateVariables(updatedConfigObject, config) {
```

Comment:
Instead of passing in `updatedConfigObject` as a mutable argument, I personally prefer this function to be side-effect free and return the updated (interpolated) object instead.

```js
function interpolateVariables(config) {
  var updatedConfigObject = {};
  // your original code
  return updatedConfigObject;
}
```